src/SonarScanner.MSBuild.Shim/TFSProcessor.Wrapper.cs

```diff
@@ -0,0 +1,107 @@
+﻿/*
+ * SonarScanner for MSBuild
+ * Copyright (C) 2016-2019 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using SonarScanner.MSBuild.Common;
+using SonarScanner.MSBuild.Shim.Interfaces;
+
+namespace SonarScanner.MSBuild.Shim
+{
+    public class TfsProcessorWrapper : ITfsProcessor
+    {
+        private readonly ILogger logger;
+
+        public TfsProcessorWrapper(ILogger logger)
+        {
+            this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        public bool Execute(AnalysisConfig config, IEnumerable<string> userCmdLineArguments, String fullPropertiesFilePath)
+        {
+            if (config == null)
+            {
+                throw new ArgumentNullException(nameof(config));
+            }
+
+            if (userCmdLineArguments == null)
+            {
+                throw new ArgumentNullException(nameof(userCmdLineArguments));
+            }
+
+            return InternalExecute(config, logger, userCmdLineArguments, fullPropertiesFilePath);
+        }
+
+        #region Private methods
+
+        private static bool InternalExecute(AnalysisConfig config, ILogger logger, IEnumerable<string> userCmdLineArguments, String fullPropertiesFilePath)
+        {
+            if (fullPropertiesFilePath == null)
+            {
+                // We expect a detailed error message to have been logged explaining
+                // why the properties file generation could not be performed
+                logger.LogInfo(Resources.MSG_PropertiesGenerationFailed);
```

Comment:
This seems weird; why does execution get this far if the properties file couldn't be generated? 
It feels odd that this isn't a normal "null ref" exception.

Reply:
Oh nice catch, that's from the old implementation, i forgot to remove this part of code. Thanks !